nemo/collections/common/tokenizers/sentencepiece_tokenizer.py

```diff
@@ -0,0 +1,259 @@
+    '''
+    Sentencepiecetokenizer https://github.com/google/sentencepiece.
+    '''
+
+    def __init__(self, model_path: str, special_tokens: Optional[Union[Dict[str, str], List[str]]] = None):
+        """
+        Args:
+            model_path: name of the pretrained model from the hugging face list
+            special_tokens: either list of special tokens or dictionary of token name to token value  
+        """
+        self.tokenizer = sentencepiece.SentencePieceProcessor()
+        self.tokenizer.Load(model_path)
+        # without special tokens
+        self.original_vocab_size = self.tokenizer.get_piece_size()
+        self.vocab_size = self.tokenizer.get_piece_size()
+        self.special_token_to_id = {}
+        self.id_to_special_token = {}
+        if special_tokens:
+            self.add_special_tokens(special_tokens)
+
+    def text_to_tokens(self, text):
+        tokens = []
+        idx = 0
+        last_idx = 0
+
+        while 1:
+            indices = {}
+
+            for token in self.special_token_to_id:
+                try:
+                    indices[token] = text[idx:].index(token)
+                except ValueError:
+                    continue
+
+            if len(indices) == 0:
+                break
+
+            next_token = min(indices, key=indices.get)
+            next_idx = idx + indices[next_token]
+
+            tokens.extend(self.tokenizer.encode_as_pieces(text[idx:next_idx]))
+            tokens.append(next_token)
+            idx = next_idx + len(next_token)
+
+        tokens.extend(self.tokenizer.encode_as_pieces(text[idx:]))
+        return tokens
+
+    def text_to_ids(self, text):
+        ids = []
+        idx = 0
+        last_idx = 0
+
+        while 1:
+            indices = {}
+
+            for token in self.special_token_to_id:
+                try:
+                    indices[token] = text[idx:].index(token)
+                except ValueError:
+                    continue
+
+            if len(indices) == 0:
+                break
+
+            next_token = min(indices, key=indices.get)
+            next_idx = idx + indices[next_token]
+
+            ids.extend(self.tokenizer.encode_as_ids(text[idx:next_idx]))
+            ids.append(self.special_token_to_id[next_token])
+            idx = next_idx + len(next_token)
+
+        ids.extend(self.tokenizer.encode_as_ids(text[idx:]))
+        return ids
+
+    def tokens_to_text(self, tokens):
+        return self.tokenizer.decode_pieces(tokens)
+
+    def ids_to_text(self, ids):
+        text = ""
+        last_i = 0
+
+        for i, id in enumerate(ids):
+            if id in self.id_to_special_token:
+                text += self.tokenizer.decode_ids(ids[last_i:i]) + " "
+                text += self.id_to_special_token[id] + " "
+                last_i = i + 1
+
+        text += self.tokenizer.decode_ids(ids[last_i:])
+        return text.strip()
+
+    def tokens_to_ids(self, tokens):
+        ids = []
```

Comment:
Could you please fix tokens_to_ids to match Nemobert input/output? It should take either str or list of strings 
https://huggingface.co/transformers/_modules/transformers/tokenization_utils_fast.html#PreTrainedTokenizerFast.convert_tokens_to_ids?

Reply:
done